src/entities/elastic-storage/elastic-storage-processor.js

```diff
@@ -598,12 +602,12 @@ function qSort(...entries) {
       if (e.length < 1 || e.length > 3) {
         throw new Error('qSort array entry expect 1-3 elements!')
       }
-      key = e[0]
+      [key] = e
       if (e.length > 1) {
-        order = e[1]
+        [, order] = e
       }
       if (e.length > 2) {
-        unmapped_type = e[2]
+        [, , unmapped_type] = e
```

Comment:
> why this is preferable to the one before it 
@rooooooooob it is just eslint rule https://eslint.org/docs/rules/prefer-destructuring

Reply:
I was curious about why that eslint rule is something desirable though. Is it for performance (less look-ups into object)? Does it do any kind of additional structure checks upon destructuring rather than silently getting undefineds everywhere in the event of a mismatch? etc. I hadn't really done much javascript until I started working on this recently so I'm curious since in some cases it seems to result in some very strange looking code when enforcing this lint rule sometimes.

Reply:
Yeah, that makes very little sense to me and I think it's harder to review. Lets just disable that rule.